### PR TITLE
TIP-706-option-options: Rework option and options normalizer and filter for es

### PR DESCRIPTION
**Pull request by momoss**
| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | OK
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Component/Catalog/Normalizer/Indexing/Product/OptionsNormalizer.php [hunk 1]
```diff
@@ -23,23 +20,14 @@ class OptionsNormalizer extends AbstractProductValueNormalizer implements Normal
      */
     public function supportsNormalization($data, $format = null)
     {
-        return $data instanceof ProductValueInterface &&
-            AttributeTypes::BACKEND_TYPE_OPTIONS === $data->getAttribute()->getBackendType() &&
-            'indexing' === $format;
+        return $data instanceof OptionsProductValue && 'indexing' === $format;
```
Comment (damien-carcel) [inline src/Pim/Component/Catalog/Normalizer/Indexing/Product/OptionsNormalizer.php]: OptionsProductValueInterface

File: src/Pim/Component/Catalog/spec/Normalizer/Indexing/Product/OptionsNormalizerSpec.php [hunk 2]
```diff
@@ -51,6 +44,31 @@ function it_support_options_product_value(
...
+                    '<all_channels>' => [
+                        '<all_locales>' => [],
+                    ],
+                ],
+            ]
+        );
+    }
+
     function it_normalize_an_options_product_value_with_no_locale_and_no_channel(
         OptionsProductValue $optionsValue,
```
Comment (damien-carcel) [inline src/Pim/Component/Catalog/spec/Normalizer/Indexing/Product/OptionsNormalizerSpec.php:71]: OptionsProductValueInterface

File: src/Pim/Component/Catalog/Normalizer/Indexing/Product/OptionNormalizer.php [hunk 3]
```diff
@@ -21,9 +20,7 @@ class OptionNormalizer extends AbstractProductValueNormalizer implements Normali
      */
     public function supportsNormalization($data, $format = null)
     {
-        return $data instanceof ProductValueInterface &&
-            AttributeTypes::BACKEND_TYPE_OPTION === $data->getAttribute()->getBackendType() &&
-            'indexing' === $format;
+        return $data instanceof OptionProductValue && 'indexing' === $format;
```
Comment (damien-carcel) [inline src/Pim/Component/Catalog/Normalizer/Indexing/Product/OptionNormalizer.php]: OptionProductValueInterface

File: src/Pim/Component/Catalog/spec/Normalizer/Indexing/Product/OptionsNormalizerSpec.php [hunk 4]
```diff
@@ -51,6 +44,31 @@ function it_support_options_product_value(
         $this->supportsNormalization($optionsValue, 'indexing')->shouldReturn(true);
     }
 
+    function it_normalize_an_empty_options_product_value(
+        OptionsProductValue $optionsValue,
```
Comment (damien-carcel) [inline src/Pim/Component/Catalog/spec/Normalizer/Indexing/Product/OptionsNormalizerSpec.php:46]: OptionsProductValueInterface

**Review by damien-carcel — APPROVED** [review]
Just a comment on interfaces.

File: src/Pim/Component/Catalog/Normalizer/Indexing/Product/OptionNormalizer.php (hunk 3 above)
Comment (skeleton) [inline src/Pim/Component/Catalog/Normalizer/Indexing/Product/OptionNormalizer.php]: OptionProductValueInterface

File: src/Pim/Component/Catalog/Normalizer/Indexing/Product/OptionsNormalizer.php (hunk 1 above)
Comment (skeleton) [inline src/Pim/Component/Catalog/Normalizer/Indexing/Product/OptionsNormalizer.php]: OptionsProductValueInterface

**Review by skeleton — APPROVED** [review]

**Pull request merged**